titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/DefaultV3JobOperations.java

```diff
@@ -608,15 +608,16 @@ public Completable recordTaskPlacement(String taskId, Function<Task, Task> chang
         Task changed = modelUpdateEvent.getChangedEntityHolder().getEntity();
         if (!modelUpdateEvent.getPreviousEntityHolder().isPresent()) {
             return tasksPredicate.test(Pair.of(job, changed))
-                    ? Optional.of(toNewTaskUpdateEvent(job, changed))
+                    ? Optional.of(toNewTaskUpdateEvent(job, changed, modelUpdateEvent.getCallMetadata()))
                     : Optional.empty();
         }
         Task previous = modelUpdateEvent.getPreviousEntityHolder().get().getEntity();
+        CallMetadata callMetadata = (CallMetadata)modelUpdateEvent.getChangedEntityHolder().getAttributes().get(JobManagerConstants.JOB_MANAGER_ATTRIBUTE_CALLMETADATA);
```

Comment:
Why the `CallMetadata` value is not taken directly from the modelUpdateEvent like above?

Reply:
I must have changed that before I used modelUpdateEvent, thanks for catching that.